cyipopt/scipy_interface.py

```diff
@@ -533,3 +538,39 @@ def minimize_ipopt(fun,
                           nfev=problem.nfev,
                           njev=problem.njev,
                           nit=problem.nit)
+
+def _minimize_ipopt_iv(fun, x0, args, kwargs, method, jac, hess, hessp,
+                       bounds, constraints, tol, callback, options):
+    # basic input validation for minimize_ipopt
+    if fun is not None and not callable(fun):
+        raise ValueError('`fun` must be callable.')
+    x0 = np.asarray(x0)[()]
+    if not np.issubdtype(x0.dtype, np.number):
+        raise ValueError('`x0` must be a numeric array.')
+    if not np.iterable(args):
+        args = (args,)
+    kwargs = dict() if kwargs is None else kwargs
+    if not isinstance(kwargs, dict):
+        raise ValueError('`kwargs` must be a dictionary.')
+    if method is not None:  # this will be updated when gh-200 is merged
```

Comment:
This and gh-200 can be merged in either order. I'll make sure this gets updated accordingly.